src/mgclient.c

```diff
@@ -379,40 +385,52 @@ static mg_map *build_hello_extra(const char *user_agent, const char *username,
     }
   }
 
-  assert((username && password) || (!username && !password));
-  if (username) {
-    mg_value *scheme = mg_value_make_string("basic");
-    if (!scheme || mg_map_insert_unsafe(extra, "scheme", scheme) != 0) {
+  // The "basic" scheme requires a username and a password/credential within the
+  // HELLO message. Other schemes (save for "kerberos", which is not supported
+  // by Memgraph) do not have such requirements:
+  // https://neo4j.com/docs/bolt/current/bolt/message/#messages-hello
+  // https://neo4j.com/docs/bolt/current/bolt/message/#messages-logon
+  if (scheme && strcmp(scheme, "basic") == 0) {
+    assert(username && password);
+  }
+
+  if (!username && !password) {
+    mg_value *scheme_ = mg_value_make_string("none");
+    if (!scheme_ || mg_map_insert_unsafe(extra, "scheme", scheme_) != 0) {
       goto cleanup;
     }
+    return extra;
+  }
+
+  mg_value *scheme_ = mg_value_make_string(scheme ? scheme : "basic");
```

Comment:
When would `scheme` return false?

Reply:
I guess when `mg_value_make_string` fails (e.g., allocation, but unlikely + when it fails since it's small, there are bigger problems most likely)?